packages/manager/apps/pci-instances/src/pages/instances/Instances.page.tsx

```diff
@@ -211,9 +211,10 @@ const Instances: FC = () => {
   );
 
   const resetSortAndFilters = useCallback(() => {
-    setSorting(initialSort);
-    filters.forEach(removeFilter);
-  }, [filters]);
+    if (filters.length) filters.forEach(removeFilter);
+    if (JSON.stringify(sorting) !== JSON.stringify(initialSorting))
```

Comment:
It seems to be a downfall on performance, could we use something else to make a deep compare ?

Reply:
@SimonChaumet Yep, agree on that but I see no other way to achieve that in a better way.
Maybe using [_isEqual](https://lodash.com/docs/4.17.15#isEqual) from Lodash, but I'm not sure about better performance, what do you think?

Reply:
You can check the tests here : https://jsperf.app/lo-dash-vs-underscore-vs-json-stringify-isequal/2

![image](https://github.com/user-attachments/assets/4cfe96ad-344d-4865-a2a7-de66f64d0c20)

Reply:
Thanks @SimonChaumet for the link:)
I also ran the test, that shows that `lodash.isEqual` is the slowest, and `underscore.isEqual` is a bit better: 
![Capture d’écran 2024-09-18 à 18 18 32](https://github.com/user-attachments/assets/30bfaf76-c530-49da-925a-2d4f030b5d27)
`JSON.stringify` seems to be in the middle, so, if you don't mind, I'll suggest to keep it like that for just such a tinny usage, instead of downloading the whole `underscore.js` library that is not declared yet in the manager.